llvm/unittests/TargetParser/TripleTest.cpp

```diff
@@ -2454,4 +2454,19 @@ TEST(TripleTest, isArmMClass) {
     EXPECT_TRUE(T.isArmMClass());
   }
 }
+
+TEST(TripleTest, DXILNormaizeWithVersion) {
+  EXPECT_EQ("dxilv1.0-unknown-shadermodel6.0",
+            Triple::normalize("dxilv1.0--shadermodel6.0"));
```

Comment:
What will Triple::normalize("dxilv1.0--shadermodel6.2") expect equal?
"dxilv1.2-unknown-shadermodel6.2"?

Reply:
> What will Triple::normalize("dxilv1.0--shadermodel6.2") expect equal? "dxilv1.2-unknown-shadermodel6.2"?

No. Since DXIL version is specified in the triple, it is honored as is. So, it will be `dxilv1.0-unknown-shadermodel6.2`

Reply:
Do we need a rule for what is legal or not?

Reply:
> So Triple::normalize("dxilv1.2--shadermodel6.0") will get "dxilv1.2-unknown-shadermodel6.0"?

Yes.

Reply:
> Do we need a rule for what is legal or not?

I expect such validation to be part of later compiler passes.